src/app/tests/BUILD.gn

```diff
@@ -54,6 +54,15 @@ source_set("binding-test-srcs") {
   ]
 }
 
+source_set("icd-manager-srcs") {
+  sources = [
+    "${chip_root}/src/app/icd/ICDManager.cpp",
```

Comment:
use a dependency instead. both src/app/BUILD.gn include these files directly and test do so ... that seems off.

Instead create one or several (in case you need ICDManager split from ICDEventManager) libs and depend on them.

Reply:
updated both gn files